core/dbt/parser/manifest.py

```diff
@@ -867,14 +898,14 @@ def read_manifest_for_partial_parse(self) -> Optional[Manifest]:
             try:
                 with open(path, "rb") as fp:
                     manifest_mp = fp.read()
-                manifest: Manifest = Manifest.from_msgpack(manifest_mp)  # type: ignore
+                manifest: Manifest = Manifest.from_msgpack(manifest_mp, decoder=extended_mashumuro_decoder)  # type: ignore
                 # keep this check inside the try/except in case something about
                 # the file has changed in weird ways, perhaps due to being a
                 # different version of dbt
                 is_partial_parsable, reparse_reason = self.is_partial_parsable(manifest)
                 if is_partial_parsable:
                     # We don't want to have stale generated_at dates
-                    manifest.metadata.generated_at = datetime.utcnow()
+                    manifest.metadata.generated_at = datetime.datetime.utcnow()
```

Comment:
`dt.utcnow()` yields a naive timestamp rather than an aware one.

Maybe we want an aware datetime like `dt.now(timezone.utc).isoformat()` instead?

☝️ assuming an import like this:
```python
from datetime import timezone, datetime as dt
```

Slightly more detail [here](https://github.com/dbt-labs/dbt-core/issues/5267#issuecomment-1513306428).

| Python                                    | Output                             |   | UTC Offset       |
|-------------------------------------------|------------------------------------|------------------|------------------|
| `dt.utcnow().isoformat()`           | `2023-04-18T13:42:18.050150`       | Naive            | N/A              |
| `dt.now().isoformat()`              | `2023-04-18T07:42:18.050150`       | Naive            | N/A              |
| `dt.now().astimezone().isoformat()` | `2023-04-18T07:42:18.050150-06:00` | Aware            | System time zone |
| `dt.now(timezone.utc).isoformat()`  | `2023-04-18T13:42:18.050150+00:00` | Aware            | UTC              |

Reply:
I am adding this to my list of Known Python Facts, but since I did not add this call, and am just changing the way the namespace is imported, I'm inclined to leave it alone for now.